pkg/logs/decoder/matcher.go

```diff
@@ -5,18 +5,76 @@
 
 package decoder
 
+var (
+	// Utf16leEOL is the bytes sequence for UTF-16 Little-Endian end-of-line char
+	Utf16leEOL = []byte{'\n', 0x00}
```

Comment:
The main reason to only catch UTF-16 encoded `\n`, and not `\r\n` is because that's what the agent is currently doing. The `\r` is removed later ([here](https://github.com/DataDog/datadog-agent/blob/7.22.0/pkg/logs/decoder/line_handler.go#L96)), this allows to keeps compatibility with line terminated by `\n` or `\r\n` even with UTF-16 files.